metadata-ingestion/tests/integration/tableau/test_tableau_ingest.py

```diff
@@ -755,30 +790,48 @@ def test_tableau_unsupported_csql(mock_datahub_graph):
     config.lineage_overrides = TableauLineageOverrides(
         database_override_map={"production database": "prod"}
     )
-    source = TableauSource(config=config, ctx=context)
-    lineage = source._create_lineage_from_unsupported_csql(
-        csql_urn="urn:li:dataset:(urn:li:dataPlatform:tableau,09988088-05ad-173c-a2f1-f33ba3a13d1a,PROD)",
-        csql={
-            "query": "SELECT user_id, source, user_source FROM (SELECT *, ROW_NUMBER() OVER (partition BY user_id ORDER BY __partition_day DESC) AS rank_ FROM invent_dw.UserDetail ) source_user WHERE rank_ = 1",
-            "isUnsupportedCustomSql": "true",
-            "database": {"name": "production database", "connectionType": "bigquery"},
-        },
-    )
 
-    mcp = cast(MetadataChangeProposalClass, next(iter(lineage)).metadata)
+    with mock.patch(
+        "datahub.ingestion.source.tableau.sqlglot_lineage"
+    ) as sqlglot_lineage:
 
-    assert mcp.aspect == UpstreamLineage(
-        upstreams=[
-            UpstreamClass(
-                dataset="urn:li:dataset:(urn:li:dataPlatform:bigquery,prod.invent_dw.userdetail,PROD)",
-                type=DatasetLineageType.TRANSFORMED,
-            )
-        ]
-    )
-    assert (
-        mcp.entityUrn
-        == "urn:li:dataset:(urn:li:dataPlatform:tableau,09988088-05ad-173c-a2f1-f33ba3a13d1a,PROD)"
-    )
+        sqlglot_lineage.return_value = SqlParsingResult(  # type:ignore
```

Comment:
why do we need to mock this - does it not produce correct results otherwise?

Reply:
In [test_tableau_cll_ingest](https://github.com/datahub-project/datahub/pull/8466/files#diff-e807b2af4d76f8f6a7e88bdd2bd32e1d58ffc7e2129e2b71eb8e4d72c84398d7R295) graph is None follow getting tested and in this test case I am providing Mock graph, so without mocking the sqlglot_lineage it returns empty array

Reply:
I don't fully follow this - let's talk about it next time we chat